### ISSUE TITLE
There are no products in the translated category

### ISSUE THREAD

**Issue by desarrollobinfor** [hyyan/woo-poly-integration#497]
#### Can you reproduce this issue on default Wordpress theme (eg Storefront)?
The website is working, I can't change the template

#### Can you reproduce this issue when all other plugins are disabled except WooCommerce, Polylang and Hyyan WooCommerce Polylang Integration?
The web is working, I can't stop the plugins.

#### What product versions and settings are you using when this issue occurs?
* PHP: 7.3.16
* WordPress: 5.3.3
* WooCommerce: 4.0.1
* Polylang: 2.7.2
* Hyyan WooCommerce Polylang Integration: 1.4.3
* Browser: All

#### Steps to Reproduce
1. See products in category (spanish): https://marketplacenewyork.com/categoria/anillos/
2. See products in same category (english): https://marketplacenewyork.com/en/categoria/rings/

#### What I Expected
The products should be the same in the category (in Spanish and English).
Spanish is the main language.

#### What Happened Instead
The same products do not appear.

## WordPress Environment

<details>
`
### WordPress Environment ###

WordPress address (URL): https://marketplacenewyork.com
Site address (URL): https://marketplacenewyork.com
WC Version: 4.0.1
REST API Version: ✔ 1.0.7
WC Blocks Version: ✔ 2.5.15
Action Scheduler Version: ✔ 3.1.4
WC Admin Version: ✔ 1.0.3
Log Directory Writable: ✔
WP Version: ❌ 5.3.3 - Hay una versión nueva de WordPress disponible (5.4.1)
WP Multisite: –
WP Memory Limit: 1 GB
WP Debug Mode: –
WP Cron: ✔
Language: es_ES
External object cache: –

### Server Environment ###

Server Info: Apache
PHP Version: 7.3.16
PHP Post Max Size: 64 MB
PHP Time Limit: 50000
PHP Max Input Vars: 5000
cURL Version: 7.38.0
OpenSSL/1.0.1t

SUHOSIN Installed: –
MySQL Version: 5.7.28-log
Max Upload Size: 64 MB
Default Timezone is UTC: ✔
fsockopen/cURL: ✔
SoapClient: ✔
DOMDocument: ✔
GZip: ✔
Multibyte String: ✔
Remote Post: ✔
Remote Get: ✔

### Database ###

WC Database Version: 4.0.1
WC Database Prefix: wp_
Tamaño total de la base de datos: 215.16MB
Tamaño de los datos de la base de datos: 134.42MB
Tamaño del índice de la base de datos: 80.74MB
wp_woocommerce_sessions: Datos: 10.02MB + Índice: 0.14MB + Motor InnoDB
wp_woocommerce_api_keys: Datos: 0.02MB + Índice: 0.03MB + Motor InnoDB
wp_woocommerce_attribute_taxonomies: Datos: 0.02MB + Índice: 0.02MB + Motor InnoDB
wp_woocommerce_downloadable_product_permissions: Datos: 0.02MB + Índice: 0.06MB + Motor InnoDB
wp_woocommerce_order_items: Datos: 2.52MB + Índice: 1.52MB + Motor InnoDB
wp_woocommerce_order_itemmeta: Datos: 10.52MB + Índice: 8.03MB + Motor InnoDB
wp_woocommerce_tax_rates: Datos: 0.02MB + Índice: 0.06MB + Motor InnoDB
wp_woocommerce_tax_rate_locations: Datos: 0.02MB + Índice: 0.03MB + Motor InnoDB
wp_woocommerce_shipping_zones: Datos: 0.02MB + Índice: 0.00MB + Motor InnoDB
wp_woocommerce_shipping_zone_locations: Datos: 0.02MB + Índice: 0.03MB + Motor InnoDB
wp_woocommerce_shipping_zone_methods: Datos: 0.02MB + Índice: 0.00MB + Motor InnoDB
wp_woocommerce_payment_tokens: Datos: 0.02MB + Índice: 0.02MB + Motor InnoDB
wp_woocommerce_payment_tokenmeta: Datos: 0.02MB + Índice: 0.03MB + Motor InnoDB
wp_woocommerce_log: Datos: 0.02MB + Índice: 0.02MB + Motor InnoDB
wp_actionscheduler_actions: Datos: 0.09MB + Índice: 0.11MB + Motor InnoDB
wp_actionscheduler_claims: Datos: 0.02MB + Índice: 0.02MB + Motor InnoDB
wp_actionscheduler_groups: Datos: 0.02MB + Índice: 0.02MB + Motor InnoDB
wp_actionscheduler_logs: Datos: 0.06MB + Índice: 0.03MB + Motor InnoDB
wp_aws_cache: Datos: 2.52MB + Índice: 0.00MB + Motor InnoDB
wp_aws_index: Datos: 6.52MB + Índice: 3.03MB + Motor InnoDB
wp_commentmeta: Datos: 0.02MB + Índice: 0.03MB + Motor InnoDB
wp_comments: Datos: 1.52MB + Índice: 0.89MB + Motor InnoDB
wp_failed_jobs: Datos: 0.02MB + Índice: 0.00MB + Motor InnoDB
wp_links: Datos: 0.02MB + Índice: 0.02MB + Motor InnoDB
wp_mailchimp_carts: Datos: 0.06MB + Índice: 0.00MB + Motor InnoDB
wp_mailchimp_jobs: Datos: 0.13MB + Índice: 0.00MB + Motor InnoDB
wp_mrw_cities: Datos: 0.02MB + Índice: 0.00MB + Motor InnoDB
wp_mrw_orders: Datos: 0.13MB + Índice: 0.00MB + Motor InnoDB
wp_mrw_ranges: Datos: 0.02MB + Índice: 0.00MB + Motor InnoDB
wp_mrw_taxes: Datos: 0.02MB + Índice: 0.00MB + Motor InnoDB
wp_options: Datos: 2.52MB + Índice: 0.16MB + Motor InnoDB
wp_postmeta: Datos: 53.58MB + Índice: 28.08MB + Motor InnoDB
wp_posts: Datos: 5.52MB + Índice: 3.80MB + Motor InnoDB
wp_queue: Datos: 0.02MB + Índice: 0.00MB + Motor InnoDB
wp_revslider_css: Datos: 0.13MB + Índice: 0.00MB + Motor InnoDB
wp_revslider_css_bkp: Datos: 0.13MB + Índice: 0.00MB + Motor InnoDB
wp_revslider_layer_animations: Datos: 0.02MB + Índice: 0.00MB + Motor InnoDB
wp_revslider_layer_animations_bkp: Datos: 0.02MB + Índice: 0.00MB + Motor InnoDB
wp_revslider_navigations: Datos: 0.02MB + Índice: 0.00MB + Motor InnoDB
wp_revslider_navigations_bkp: Datos: 0.02MB + Índice: 0.00MB + Motor InnoDB
wp_revslider_sliders: Datos: 0.16MB + Índice: 0.00MB + Motor InnoDB
wp_revslider_sliders_bkp: Datos: 0.27MB + Índice: 0.00MB + Motor InnoDB
wp_revslider_slides: Datos: 1.52MB + Índice: 0.00MB + Motor InnoDB
wp_revslider_slides_bkp: Datos: 1.52MB + Índice: 0.00MB + Motor InnoDB
wp_revslider_static_slides: Datos: 0.03MB + Índice: 0.00MB + Motor InnoDB
wp_revslider_static_slides_bkp: Datos: 0.05MB + Índice: 0.00MB + Motor InnoDB
wp_sbi_instagram_feeds_posts: Datos: 0.02MB + Índice: 0.02MB + Motor InnoDB
wp_sbi_instagram_posts: Datos: 0.09MB + Índice: 0.00MB + Motor InnoDB
wp_termmeta: Datos: 0.06MB + Índice: 0.03MB + Motor InnoDB
wp_terms: Datos: 0.14MB + Índice: 0.19MB + Motor InnoDB
wp_term_relationships: Datos: 1.52MB + Índice: 1.52MB + Motor InnoDB
wp_term_taxonomy: Datos: 0.20MB + Índice: 0.19MB + Motor InnoDB
wp_usermeta: Datos: 19.55MB + Índice: 25.09MB + Motor InnoDB
wp_users: Datos: 1.52MB + Índice: 1.13MB + Motor InnoDB
wp_wcpdf_invoice_number: Datos: 0.02MB + Índice: 0.00MB + Motor InnoDB
wp_wc_admin_notes: Datos: 0.02MB + Índice: 0.00MB + Motor InnoDB
wp_wc_admin_note_actions: Datos: 0.02MB + Índice: 0.02MB + Motor InnoDB
wp_wc_category_lookup: Datos: 0.02MB + Índice: 0.00MB + Motor InnoDB
wp_wc_customer_lookup: Datos: 1.52MB + Índice: 0.52MB + Motor InnoDB
wp_wc_download_log: Datos: 0.02MB + Índice: 0.03MB + Motor InnoDB
wp_wc_order_coupon_lookup: Datos: 0.02MB + Índice: 0.03MB + Motor InnoDB
wp_wc_order_product_lookup: Datos: 0.08MB + Índice: 0.06MB + Motor InnoDB
wp_wc_order_stats: Datos: 1.52MB + Índice: 0.64MB + Motor InnoDB
wp_wc_order_tax_lookup: Datos: 0.48MB + Índice: 0.38MB + Motor InnoDB
wp_wc_product_meta_lookup: Datos: 1.52MB + Índice: 0.92MB + Motor InnoDB
wp_wc_tax_rate_classes: Datos: 0.02MB + Índice: 0.02MB + Motor InnoDB
wp_wc_webhooks: Datos: 0.02MB + Índice: 0.02MB + Motor InnoDB
wp_yith_wcwl: Datos: 2.52MB + Índice: 0.41MB + Motor InnoDB
wp_yith_wcwl_lists: Datos: 3.52MB + Índice: 3.34MB + Motor InnoDB

### Post Type Counts ###

attachment: 3055
custom_css: 2
fgf_master_log: 35
fgf_rules: 1
nav_menu_item: 80
oembed_cache: 1
page: 69
polylang_mo: 2
post: 2
product: 2904
product_variation: 1593
shop_coupon: 255
shop_order: 7247
shop_order_refund: 6
shop-masonry-gallery: 21
wcct_countdown: 2
wpcf7_contact_form: 4

### Security ###

Secure connection (HTTPS): ✔
Hide errors from visitors: ✔

### Active Plugins (33) ###

Polylang: por WP SYNTEX – 2.7.2
Advanced Woo Search: por ILLID – 2.00
Contact Form 7 Extension For Mailchimp: por Renzo Johnson – 0.5.17
Contact Form 7: por Takayuki Miyoshi – 5.1.7
Cookie Notice: por dFactory – 1.3.1
Envato Market: por Envato – 2.0.3
Finale Lite - Sales Countdown Timer & Discount for WooCommerce: por XLPlugins – 2.12.0
Free Gifts for WooCommerce: por FantasticPlugins – 4.3
Smash Balloon Instagram Feed: por Smash Balloon – 2.4
WPBakery Page Builder: por Michael M - WPBakery.com – 6.1
Mailchimp for WooCommerce: por Mailchimp – 2.4.0
Onea Core: por Elated Themes – 1.2
Onea Membership: por Elated Themes – 1.0
Onea Twitter Feed: por Elated Themes – 1.0.1
Product Assembly / Gift Wrap / ... Cost for WooCommerce: por Webdados – 1.2
Quick remove menu item: por hpthemes – 0.1
Really Simple SSL: por Rogier Lankhorst
Mark Wolters – 3.3.1

Redsys WooCommerce: por Redsys – 3.0.1 – No probado con la versión activa de WooCommerce
Slider Revolution: por ThemePunch – 6.1.5
WooCommerce Blocks: por Automattic – 2.5.15
Advanced Order Export For WooCommerce: por AlgolPlus – 3.1.4
Hyyan WooCommerce Polylang Integration: por Hyyan Abo Fakher – 1.4.3 – No probado con la versión activa de WooCommerce
WooCommerce PayPal Checkout Gateway: por WooCommerce – 1.6.21
WooCommerce Google Analytics Integration: por WooCommerce – 1.4.20
MRW: por Webubi – 4.1.8
WooCommerce PDF Invoices & Packing Slips: por Ewout Fernhout – 2.4.9
WooCommerce: por Automattic – 4.0.1
WP Fastest Cache Premium: por Emre Vona – 1.5.7
WP Fastest Cache: por Emre Vona – 0.9.0.5
WP Mail SMTP: por WPForms – 2.0.0
YITH WooCommerce Quick View: por YITH – 1.4.1
YITH WooCommerce Waiting List: por YITH – 1.4.1
YITH WooCommerce Wishlist: por YITH – 3.0.9

### Inactive Plugins (2) ###

FG PrestaShop to WooCommerce Premium: por Frédéric GILLES – 3.64.0 – No probado con la versión activa de WooCommerce
Onea Instagram Feed: por Elated Themes – 1.0.1

### Settings ###

API Enabled: –
Force SSL: –
Currency: EUR (€)
Currency Position: right
Thousand Separator: .
Decimal Separator: ,
Number of Decimals: 2
Taxonomies: Product Types: simple (simple)
variable (variable)

Taxonomies: Product Visibility: outofstock (outofstock)
Connected to WooCommerce.com: ✔

### WC Pages ###

Base de la tienda: #7049 - /tienda/
Carrito: #6 - /carrito/
Finalizar compra: #7 - /finalizar-compra/
Mi cuenta: #8 - /mi-cuenta/
Términos y condiciones: #8358 - /terminos-y-condiciones/

### Theme ###

Name: Onea Child Theme
Version: 1.4
Author URL: http://www.binfor.es
Child Theme: ✔
Parent Theme Name: Onea
Parent Theme Version: 1.4
Parent Theme Author URL: http://themeforest.net/user/elated-themes
WooCommerce Support: ✔

### Templates ###

Archive Template: Tu tema tiene un archivo woocommerce.php
no podrás sobreescribir la plantilla personalizada woocommerce/archive-product.php ya que woocommerce.php tiene prioridad sobre archive-product.php. Esto es intencionado para prevenir problemas de visualización.

Overrides: onea/woocommerce/cart/cart.php
onea/woocommerce/content-product.php
La  version 3.6.0 de onea/woocommerce/global/quantity-input.php está obsoleta. La versión del núcleo es la 4.0.0
onea/woocommerce/product-searchform.php

Outdated Templates: ❌
					
					
						Aprende cómo actualizar

### Action Scheduler ###

Complete: 370
Oldest: 2020-04-28 08:54:59 +0200
Newest: 2020-04-30 16:17:25 +0200

`
</details>

**Comment by Jon007** [comment]
Possible problem:

1)
Try turning off or clearing WP Fastest Cache or other plugins.
My guess is that a caching layer has cached an old version of the category pages before the other English translations were published.

For example when using Redis caching I found it is specifically necessary to exclude 'product_cat' from caching.

Although WooCommerce 4.x is not explicitly supported by released versions of this plugin I am testing 4.2.2 and not seeing this type of problem.

2) 
go to WooCommerce, Products, Categories, normally /wp-admin/edit-tags.php?taxonomy=product_cat&post_type=product
and check:
- are there any categories with duplicate slug?  This could break the system.
- are the categories correctly linked to their translations

if so, it should work, but if not please reopen with more details.

**Comment by desarrollobinfor** [comment]
Good morning,

I've checked the following:
- The categories are well translated. No duplicate slugs.
- I cleared all cache and disabled the cache plugin. (After the tests I activated it again).
But all the products in the categories are still missing.

For example:
- Anillos (Spanish) has 94 products https://marketplacenewyork.com/categoria/anillos/
- Rings (English) has 9 products https://marketplacenewyork.com/en/categoria/rings/

Fails mainly with variable products. Although sometimes it does not show some simple ones either.
What can happen?

Thank you very much.
